### DISCUSSION TITLE
Need help getting started

### DISCUSSION THREAD

**Discussion by careingemt** [tagyoureit/nodejs-poolController#771] [Nixie] [answered]
Hello,
  Trying to get nodejs running on my pi .. I have ran the "Raspberry Pi Quick Start" 

However :   86 packages are looking for funding still ... is this normal?

### fran@poolpi:~/nodejs-poolController-dashPanel $ cd ~

git clone https://github.com/rstrouse/relayEquipmentManager.git

cd ~/relayEquipmentManager

npm i
fatal: destination path 'relayEquipmentManager' already exists and is not an empty directory.
npm WARN deprecated eslint-plugin-standard@5.0.0: standard 16.0.0 and eslint-config-standard 16.0.0 no longer require the eslint-plugin-standard package. You can remove it from your dependencies with 'npm rm eslint-plugin-standard'. More info here: https://github.com/standard/standard/issues/1316

added 412 packages, and audited 413 packages in 2m

86 packages are looking for funding
  run `npm fund` for details

2 vulnerabilities (1 moderate, 1 high)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
fran@poolpi:~/relayEquipmentManager $ npm audit fix

changed 2 packages, and audited 413 packages in 11s

86 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
###

**Comment by rstrouse** [comment]
Are you running with Remote Desktop Connection or VNC?

**Reply by careingemt** [reply]
Remote Desktop Connection

**Reply by rstrouse** [reply]
If you have not implemented PM2 yet open a console then type
```
cd ~/nodejs-poolController
npm start
```

Then open another console without closing the first one and type
```
cd ~/nodejs-poolController-dashPanel
npm start
```

Then open another console without closing the other two and type
```
cd ~/relayEquipmentManager
npm start
```

Then from a browser window enter the following url.  `http://<ip address of the pi>:5150`

**Comment by careingemt** [comment]
![55](https://github.com/tagyoureit/nodejs-poolController/assets/36652951/4f0f955d-7478-4024-ac5f-402e24935c54)
OH WOW !!

**Reply by rstrouse** [reply]
Now click on the connections tab and introduce dashPanel to the njsPC server.  If this is all on the same pi you can use the loopback address if it is set up.

**Comment by careingemt** [comment]
If this is all on the same pi you can use the loopback address if it is set up. it is 1 pi but not sure what you mean by loop back

**Comment by careingemt** [comment]
![56](https://github.com/tagyoureit/nodejs-poolController/assets/36652951/26bc9c31-18db-41ca-aa9f-9bd4eb9ef6da)

so this is what i have after clicking connections then click apply

**Reply by tagyoureit** [reply]
What pool equipment (control panel, pumps, etc) do you have?

**Comment by careingemt** [comment]
raspberry pi
mechanical relay which closes a contactor to run a  hayward single speed pump
flow switch: which tells the pi  (python script, open or closed) if the pump is moving water
DS18B20 temperature probe (floating in my pool NOT plumbed) 
AR-PRO 520272 Air/Water/Solar Temperature Sensor will arrive on tomorrow

**Reply by tagyoureit** [reply]
Gotcha - so you don't have an OCP (outdoor control panel) and hence the Nixie single body is what you want.  

Sounds like you have the contactor, but just for reference we recommend 3HP relays ([High current circuits](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#high-current-circuits)).  You need to add your pump in the configuration section (gears in the upper right corner) and assign one (or more) circuits to the pump.  When you turn on the circuit it will turn on the pump relay.  

You'll also need [REM](https://github.com/rstrouse/relayEquipmentManager) to turn on/off the relays.  REM will control all of your physical equipment (relays, temp probes).  You'll configure the relay in REM and then tell njsPC to use the relay to turn on the relay through REM.  [Connect njsPC to REM](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#configuring-the-software).

**Reply by rstrouse** [reply]
Do you have an ADC such as an ADS11x.  If you do wire it to your pi then open `http://<ip address of pi>:8080` this will allow you to configure i2c, spi, and gpio devices attached to your system.

Here is a wiki on setting up Nixie
https://github.com/tagyoureit/nodejs-poolController/wiki/Nixie-Controller-Setup

Here is a wiki on setting up REM (Relay Equipment Manager) in general
https://github.com/rstrouse/relayEquipmentManager/wiki

Here is a wiki on setting up temperature sensors in REM (Relay Equipment Manager)
https://github.com/rstrouse/relayEquipmentManager/wiki/Temperature-Sensors

**Reply by careingemt** [reply]
ADC /ADS11x .... no idea what that is

**Reply by careingemt** [reply, marked as answer]
no idea how to install NIXIE

**Reply by tagyoureit** [reply]
Nixie is the name for the software based pool controller that replaces a Pentair, Aqualink, etc OCP (outdoor control panel).  It is installed with njsPC and will be the default control panel if no rs-485 connections are found that are connected to another said controller.

An ADC is an analog/digital converter.  It takes analog signals and converts then to digital ones for the Pi (or other computer) to read.  The ADS111x is a family of ADC chips.  Any one of them will do, but there are some underlying technical differences in how fast they can convert signals and to what precision.

Your DS18B20 is 1-wire which allows it to communicate digitally and natively.  However, the AR-PRO replacement sensors are analog sensors and in order to use them you need the ADC to convert the signals into digital ones.
